<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>570</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Maze</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>20</y>
      <width>281</width>
      <height>521</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="cave">
     <attribute name="title">
      <string>Лабиринт</string>
     </attribute>
     <widget class="QWidget" name="verticalLayoutWidget_2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>271</width>
        <height>182</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QPushButton" name="open_maze_btn">
           <property name="text">
            <string>Открыть</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="clear_maze">
           <property name="text">
            <string>Очистить</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="save_maze_button">
         <property name="text">
          <string>Сохранить текущий лабиринт</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Rows</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="rows_spin">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>50</number>
             </property>
             <property name="singleStep">
              <number>1</number>
             </property>
             <property name="value">
              <number>5</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Colums</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="cols_spin">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>50</number>
             </property>
             <property name="value">
              <number>5</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="generate_maze_btn">
           <property name="text">
            <string>Генерировать новый лабиринт</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="find_way_btn">
           <property name="text">
            <string>Найти путь между двумя точками</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="menu_2">
     <attribute name="title">
      <string>Пещера</string>
     </attribute>
     <widget class="QWidget" name="verticalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>261</width>
        <height>321</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_10">
         <item>
          <widget class="QPushButton" name="open_cave_btn">
           <property name="text">
            <string>Open</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="clear_cave_btn">
           <property name="text">
            <string>Clear</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>Вероятность рождения: </string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="cave_percent_spinBox">
           <property name="maximum">
            <number>100</number>
           </property>
           <property name="value">
            <number>50</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>%</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>Rows</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="rows_cave_spin">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>50</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>Colums</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="cols_cave_spin">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>50</number>
           </property>
           <property name="value">
            <number>5</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="init_cave_btn">
         <property name="text">
          <string>Инициализировать пешеру</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>Рождения</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinBirth">
           <property name="maximum">
            <number>8</number>
           </property>
           <property name="value">
            <number>3</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Смерти</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="spinDeath">
           <property name="maximum">
            <number>8</number>
           </property>
           <property name="value">
            <number>4</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_9">
         <item>
          <widget class="QRadioButton" name="auto_radioButton">
           <property name="text">
            <string>Авто</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="manual_radioButton_2">
           <property name="text">
            <string>Вручную</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QLabel" name="label_12">
           <property name="text">
            <string>Время шага:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="time_spinBox">
           <property name="maximum">
            <number>1000</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_11">
           <property name="text">
            <string>ms</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="QLabel" name="label_13">
           <property name="text">
            <string>Количество шагов</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="steps_spinBox_2">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>20</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="generate_cave_btn">
         <property name="text">
          <string>Generate</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>505</y>
      <width>491</width>
      <height>16</height>
     </rect>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>520</y>
      <width>273</width>
      <height>32</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Widht_line</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="widht_line_spin">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>10</number>
       </property>
       <property name="value">
        <number>2</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Color line</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="color_lines_comboBox">
       <property name="currentText">
        <string>RED</string>
       </property>
       <item>
        <property name="text">
         <string>RED</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>GREEN</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>BLUE</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>BLACK</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="maze_open">
   <property name="text">
    <string>Открыть лабиринт</string>
   </property>
  </action>
  <action name="cave_open">
   <property name="text">
    <string>Открыть пещеру</string>
   </property>
  </action>
  <action name="action">
   <property name="text">
    <string>Выход</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
